queryhelper/src/main/java/queryhelper/service/AIService.java

```diff
@@ -0,0 +1,117 @@
+package queryhelper.service;
+
+import com.adaptivescale.rosetta.common.DriverManagerDriverProvider;
+import com.adaptivescale.rosetta.common.JDBCUtils;
+import com.adaptivescale.rosetta.common.models.input.Connection;
+import com.adataptivescale.rosetta.source.common.QueryHelper;
+import com.google.gson.Gson;
+import dev.langchain4j.model.openai.OpenAiChatModel;
+import net.sf.jsqlparser.parser.CCJSqlParserUtil;
+import net.sf.jsqlparser.statement.select.Select;
+import queryhelper.pojo.GenericResponse;
+import queryhelper.pojo.QueryDataResponse;
+import queryhelper.pojo.QueryRequest;
+import queryhelper.utils.ErrorUtils;
+import queryhelper.utils.FileUtils;
+import queryhelper.utils.PromptUtils;
+
+import java.sql.Driver;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+
+public class AIService {
+    private final static String AI_MODEL = "gpt-3.5-turbo";
+    public static GenericResponse generateQuery(QueryRequest queryRequest, String apiKey, String aiModel, String databaseDDL, Connection source, String csvFileName, Integer showRowLimit) {
+
+        Gson gson = new Gson();
+        GenericResponse response = new GenericResponse();
+        QueryDataResponse data = new QueryDataResponse();
+
+        long startTime;
+        long endTime;
+        double elapsedTime;
+        String query;
+        String aiOutputStr;
+        String prompt;
+
+        OpenAiChatModel.OpenAiChatModelBuilder model = OpenAiChatModel
+                .builder()
+                .temperature(0.1)
```

Comment:
Any reason why we are not using temperature value **0.0**? I have seen that **0.1** works great so far.

Reply:
As far as making the model more deterministic, this needs to be tested, with the temperature 0.1 the prompt has been providing very solid responses, tweaking the prompt may be something that requires more testing, but as it is right now we are not facing any problems